File: flut/.flutter/lib/flutter/rendering/flex.dart
import 'package:flutter/material.dart';
import 'package:flut/flut/object.dart';

class FlutMainAxisSize extends FlutEnumObject<MainAxisSize> {
  const FlutMainAxisSize()
    : super('MainAxisSize', const {
        'min': MainAxisSize.min,
        'max': MainAxisSize.max,
      });
}

class FlutMainAxisAlignment extends FlutEnumObject<MainAxisAlignment> {
  const FlutMainAxisAlignment()
    : super('MainAxisAlignment', const {
        'start': MainAxisAlignment.start,
        'end': MainAxisAlignment.end,
        'center': MainAxisAlignment.center,
        'spaceBetween': MainAxisAlignment.spaceBetween,
        'spaceAround': MainAxisAlignment.spaceAround,
        'spaceEvenly': MainAxisAlignment.spaceEvenly,
      });
}

class FlutCrossAxisAlignment extends FlutEnumObject<CrossAxisAlignment> {
  const FlutCrossAxisAlignment()
    : super('CrossAxisAlignment', const {
        'start': CrossAxisAlignment.start,
        'end': CrossAxisAlignment.end,
        'center': CrossAxisAlignment.center,
        'stretch': CrossAxisAlignment.stretch,
        'baseline': CrossAxisAlignment.baseline,
      });
}

class FlutFlexFit extends FlutEnumObject<FlexFit> {
  const FlutFlexFit()
    : super('FlexFit', const {'tight': FlexFit.tight, 'loose': FlexFit.loose});
}

File: flut/flutter/rendering/flex.py
from flut._flut.engine import FlutEnumObject


class MainAxisSize(FlutEnumObject):
    min: "MainAxisSize"
    max: "MainAxisSize"


class MainAxisAlignment(FlutEnumObject):
    start: "MainAxisAlignment"
    end: "MainAxisAlignment"
    center: "MainAxisAlignment"
    spaceBetween: "MainAxisAlignment"
    spaceAround: "MainAxisAlignment"
    spaceEvenly: "MainAxisAlignment"


class CrossAxisAlignment(FlutEnumObject):
    start: "CrossAxisAlignment"
    end: "CrossAxisAlignment"
    center: "CrossAxisAlignment"
    stretch: "CrossAxisAlignment"
    baseline: "CrossAxisAlignment"


class FlexFit(FlutEnumObject):
    tight: "FlexFit"
    loose: "FlexFit"
